Tests/kaas/k8s-node-distribution/k8s-node-distribution-check.py

```diff
@@ -0,0 +1,258 @@
+            config.kubeconfig = opt[1]
+
+    return config
+
+
+def setup_logging(config_log):
+
+    logging.config.dictConfig(config_log)
+    loggers = [
+        logging.getLogger(name)
+        for name in logging.root.manager.loggerDict
+        if not logging.getLogger(name).level
+    ]
+
+    for log in loggers:
+        log.setLevel(config_log['level'])
+
+
+def initialize_config(config):
+    """Initialize the configuration for the test script"""
+
+    try:
+        with open(config.config_path, "r") as f:
+            config.logging = yaml.safe_load(f)['logging']
+    except OSError:
+        logger.warning(f"The config file under {config.config_path} couldn't be found, "
+                       f"falling back to the default config.")
+    finally:
+        # Setup logging if the config file with the relevant information could be loaded before
+        # Otherwise, we initialize logging with the included literal
+        setup_logging(config.logging or logging_config)
+
+    if config.kubeconfig is None:
+        raise ConfigException("A kubeconfig needs to be set in order to test a k8s cluster version.")
+
+    return config
+
+
+async def get_k8s_cluster_labelled_nodes(kubeconfig, labels):
+    """Get the labels of the k8s cluster under test."""
+    _ = await kubernetes_asyncio.config.load_kube_config(kubeconfig)
+
+    nodes = list()
+
+    async with kubernetes_asyncio.client.ApiClient() as api:
+        core_api = kubernetes_asyncio.client.CoreV1Api(api)
+        result = await core_api.list_node()
+        for item in result.items:
+            node = dict()
+            node_labels = item.metadata.labels
+            for label in labels:
+                try:
+                    node[label] = node_labels[label]
+                except KeyError:
+                    # Node is not labelled as a control plane node
+                    pass
+            nodes.append(node)
+
+    return nodes
+
+
+def compare_labels(node_list, labels, node_type="master"):
+
+    label_data = dict()
+
+    for node in node_list:
+        for key in labels:
+            if key not in label_data:
+                label_data[key] = set()
+            try:
+                label_data[key].add(node[key])
+            except KeyError:
+                logger.warning(f"The label for {labels[key].split('/')[1]}s don't seem to be set for all nodes.")
+
+    for label in labels:
+        if len(labels[label]) <= 1:
+            if len(labels[label]) == 1:
+                logger.warning(f"There seems to be no distribution across multiple {label.split('/')[1]}s "
+                               "or labels aren't set correctly across nodes.")
+            elif len(labels[label]) == 0:
+                logger.warning(f"The label for {label.split('/')[1]}s don't seem to be set for all nodes.")
+        else:
+            logger.info(f"The nodes are distributed across {str(len(labels[label]))} {label.split('/')[1]}s.")
+            return
```

Comment:
Yes the `labels` needed to be replaced with `label_data`, already did that now.

I'm just confused about the second part, which comment about these lines? Or did these already switch line?

Reply:
I replied to the comment in question.

Reply:
Ah, you're right, of course! We could use a `collections.Counter` instead of a set; this should cover both use cases. But I think we can postpone this. I don't know how important this warning is anyway.